app/controllers/users_controller.rb

```diff
@@ -1,7 +1,16 @@
 class UsersController < ApplicationController
-  before_action :load_user, only: :show
+  before_action :logged_in_user, except: %i(new show create)
+  before_action :correct_user, only: %i(edit update)
+  before_action :admin_user, only: :destroy
 
-  def show; end
+
+  def index
+    @users = User.paginate(page: params[:page])
+  end
+
+  def show
+    @user = User.find_by(id: params[id:])
```

Comment:
if you remove "before_action :load_user, only: :show" you need handle in this
```suggestion
    @user = User.find_by(id: params[id:])
    return if @user
    
    flash[:error] = t(:error)
    redirect_to root_path    
```